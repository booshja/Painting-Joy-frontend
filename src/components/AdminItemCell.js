// dependencies
import React from "react";
import styled from "styled-components";
// components
import {
    StyledGreenSoloButton,
    StyledOutlineButton,
} from "../containers/styles/adminButtons";
import { StyledP } from "../containers/styles/adminTypography";
import { StyledCell } from "../containers/styles/adminContainers";
// hooks
import { useHistory } from "react-router";

const StyledInlineBlockP = styled(StyledP)`
    display: inline-block;
    word-break: break-word;
    margin-bottom: 0.5rem;
`;

const StyledBoldP = styled(StyledInlineBlockP)`
    font-weight: 700;
`;

const StyledButtonContainer = styled.div`
    display: flex;
    justify-content: space-between;
    width: ${(props) => (props.showSoldOut ? "45%" : "100%")};
    align-self: flex-end;
    margin: 1rem 0 0;
`;

const AdminItemCell = ({
    data,
    handleMarkSoldOut,
    handleDelete,
    showSoldOut,
}) => {
    // set up history
    const history = useHistory();
    // destructure data
    const { id, name, shipping, price, description, quantity, isSold } = data;

    return (
        <StyledCell>
            <StyledBoldP>{name}</StyledBoldP>
            <StyledInlineBlockP>{description}</StyledInlineBlockP>
            {!isSold && (
                <StyledInlineBlockP>Quantity: {quantity}</StyledInlineBlockP>
            )}
            {isSold && <StyledBoldP>Sold Out!</StyledBoldP>}
            <StyledInlineBlockP>Price: ${price}</StyledInlineBlockP>
            <StyledInlineBlockP>Shipping: ${shipping}</StyledInlineBlockP>
            <StyledInlineBlockP>
                Total: ${(+shipping + +price).toFixed(2)}
            </StyledInlineBlockP>
            <StyledButtonContainer showSoldOut={showSoldOut}>
                {showSoldOut ? (
                    <>
                        <StyledOutlineButton
                            color="#a1c9c9"
                            onClick={() =>
                                history.push(`/admin/items/edit/${id}`)
                            }
                        >
                            Edit
                        </StyledOutlineButton>
                        <StyledOutlineButton
                            color="#db9487"
                            onClick={() => handleDelete(id)}
                        >
                            Delete
                        </StyledOutlineButton>
                    </>
                ) : (
                    <>
                        <StyledOutlineButton
                            color="#a1c9c9"
                            onClick={() =>
                                history.push(`/admin/items/edit/${id}`)
                            }
                        >
                            Edit
                        </StyledOutlineButton>
                        <StyledGreenSoloButton
                            onClick={() => handleMarkSoldOut(id)}
                        >
                            Mark Sold Out
                        </StyledGreenSoloButton>
                        <StyledOutlineButton
                            color="#db9487"
                            onClick={() => handleDelete(id)}
                        >
                            Delete
                        </StyledOutlineButton>
                    </>
                )}
            </StyledButtonContainer>
        </StyledCell>
    );
};

export default AdminItemCell;
