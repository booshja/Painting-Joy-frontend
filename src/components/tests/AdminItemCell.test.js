import { render } from "@testing-library/react";
import AdminItemCell from "../AdminItemCell";

/** Smoke Test */
it("renders without crashing, showSoldOut true, isSold false", () => {
    render(
        <AdminItemCell
            data={{
                id: 1,
                name: "Test Name",
                shipping: "123.99",
                price: "299.99",
                description: "Description",
                quantity: 3,
                isSold: false,
            }}
            handleMarkSoldOut={null}
            handleDelete={null}
            showSoldOut={true}
        />
    );
});

it("renders without crashing, showSoldOut true, isSold true", () => {
    render(
        <AdminItemCell
            data={{
                id: 2,
                name: "Test Name 2",
                shipping: "123.99",
                price: "299.99",
                description: "Description 2",
                quantity: 0,
                isSold: true,
            }}
            handleMarkSoldOut={null}
            handleDelete={null}
            showSoldOut={true}
        />
    );
});

it("renders without crashing, showSoldOut false, isSold false", () => {
    render(
        <AdminItemCell
            data={{
                id: 3,
                name: "Test Name 3",
                shipping: "98.98",
                price: "333.55",
                description: "Description 3",
                quantity: 5,
                isSold: false,
            }}
            handleMarkSoldOut={null}
            handleDelete={null}
            showSoldOut={false}
        />
    );
});

it("renders without crashing, showSoldOut false, isSold true", () => {
    render(
        <AdminItemCell
            data={{
                id: 4,
                name: "Test Name 4",
                shipping: "678.44",
                description: "Description 4",
                price: "456.66",
                quantity: 0,
                isSold: true,
            }}
            handleMarkSoldOut={null}
            handleDelete={null}
            showSoldOut={false}
        />
    );
});

/** Snapshot Test */
it("matches snapshot, showSoldOut true, isSold false", () => {
    const { asFragment } = render(
        <AdminItemCell
            data={{
                id: 1,
                name: "Test Name",
                shipping: "123.99",
                price: "299.99",
                quantity: 3,
                isSold: false,
            }}
            handleMarkSoldOut={null}
            handleDelete={null}
            showSoldOut={true}
        />
    );
    expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot, showSoldOut true, isSold true", () => {
    const { asFragment } = render(
        <AdminItemCell
            data={{
                id: 2,
                name: "Test Name 2",
                shipping: "123.99",
                price: "299.99",
                quantity: 0,
                isSold: true,
            }}
            handleMarkSoldOut={null}
            handleDelete={null}
            showSoldOut={true}
        />
    );
    expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot, showSoldOut false, isSold false", () => {
    const { asFragment } = render(
        <AdminItemCell
            data={{
                id: 3,
                name: "Test Name 3",
                shipping: "98.98",
                price: "333.55",
                quantity: 5,
                isSold: false,
            }}
            handleMarkSoldOut={null}
            handleDelete={null}
            showSoldOut={false}
        />
    );
    expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot, showSoldOut false, isSold true", () => {
    const { asFragment } = render(
        <AdminItemCell
            data={{
                id: 4,
                name: "Test Name 4",
                shipping: "678.44",
                price: "456.66",
                quantity: 0,
                isSold: true,
            }}
            handleMarkSoldOut={null}
            handleDelete={null}
            showSoldOut={false}
        />
    );
    expect(asFragment()).toMatchSnapshot();
});

/** RTL Tests */
it("renders correctly, showSoldOut true, isSold false", () => {
    const { getByText, getByRole, getByAltText } = render(
        <AdminItemCell
            data={{
                id: 1,
                name: "Test Name",
                description: "Description",
                shipping: "123.99",
                price: "299.99",
                quantity: 3,
                isSold: false,
            }}
            handleMarkSoldOut={null}
            handleDelete={null}
            showSoldOut={true}
        />
    );

    getByText("Test Name");
    getByText("Description");
    getByText("Quantity: 3");
    getByText("Price: $299.99");
    getByText("Shipping: $123.99");
    getByText("Total: $423.98");
    getByRole("button", { name: "Edit" });
    getByText("Mark Sold Out");
    getByRole("button", { name: "Delete" });
    getByAltText("Sale Item");
});

it("renders correctly, showSoldOut false, isSold true", () => {
    const { getByText, getByRole, getByAltText } = render(
        <AdminItemCell
            data={{
                id: 4,
                name: "Test Name 4",
                shipping: "678.44",
                price: "456.66",
                description: "Description 4",
                quantity: 0,
                isSold: true,
            }}
            handleMarkSoldOut={null}
            handleDelete={null}
            showSoldOut={false}
        />
    );

    getByText("Test Name 4");
    getByText("Description 4");
    getByText("Sold Out!");
    getByText("Price: $456.66");
    getByText("Shipping: $678.44");
    getByText("Total: $1135.10");
    getByRole("button", { name: "Edit" });
    getByRole("button", { name: "Delete" });
    getByAltText("Sale Item");
});

it("renders correctly, showSoldOut true, isSold true", () => {
    const { getByText, getByRole, getByAltText } = render(
        <AdminItemCell
            data={{
                id: 2,
                name: "Test Name 2",
                shipping: "123.99",
                description: "Description 2",
                price: "299.99",
                quantity: 0,
                isSold: true,
            }}
            handleMarkSoldOut={null}
            handleDelete={null}
        />
    );

    getByText("Test Name 2");
    getByText("Description 2");
    getByText("Sold Out!");
    getByText("Price: $299.99");
    getByText("Shipping: $123.99");
    getByText("Total: $423.98");
    getByRole("button", { name: "Edit" });
    getByRole("button", { name: "Delete" });
    getByAltText("Sale Item");
});

it("renders correctly, showSoldOut false, isSold false", () => {
    const { getByText, getByRole, getByAltText } = render(
        <AdminItemCell
            data={{
                id: 3,
                name: "Test Name 3",
                shipping: "98.98",
                price: "333.55",
                description: "Description 3",
                quantity: 5,
                isSold: false,
            }}
            handleMarkSoldOut={null}
            handleDelete={null}
            showSoldOut={false}
        />
    );

    getByText("Test Name 3");
    getByText("Description 3");
    getByText("Quantity: 5");
    getByText("Price: $333.55");
    getByText("Shipping: $98.98");
    getByText("Total: $432.53");
    getByRole("button", { name: "Edit" });
    getByText("Mark Sold Out");
    getByRole("button", { name: "Delete" });
    getByAltText("Sale Item");
});
