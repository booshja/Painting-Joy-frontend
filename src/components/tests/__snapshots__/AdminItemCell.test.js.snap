// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot, showSoldOut false, isSold false 1`] = `
<DocumentFragment>
  <div
    class="sc-hKFxyN sc-eCApnc kpxIQl inHhUO"
  >
    <div
      class="sc-kEqXSa bXyTKs"
    >
      <p
        class="sc-dlnjwi sc-jSFjdj sc-gKAaRy dHRUYW hDOwCU kzLjJL"
      >
        Test Name 3
      </p>
      <p
        class="sc-dlnjwi sc-jSFjdj dHRUYW hDOwCU"
      />
      <p
        class="sc-dlnjwi sc-jSFjdj dHRUYW hDOwCU"
      >
        Quantity: 5
      </p>
      <p
        class="sc-dlnjwi sc-jSFjdj dHRUYW hDOwCU"
      >
        Price: $333.55
      </p>
      <p
        class="sc-dlnjwi sc-jSFjdj dHRUYW hDOwCU"
      >
        Shipping: $98.98
      </p>
      <p
        class="sc-dlnjwi sc-jSFjdj dHRUYW hDOwCU"
      >
        Total: $432.53
      </p>
      <div
        class="sc-iCoGMd gCnnPv"
      >
        <button
          class="sc-gtsrHT qvdyu"
          color="#a1c9c9"
        >
          Edit
        </button>
        <button
          class="sc-bdnxRM sc-pNWdM gcBejj kbzcVe"
        >
          Mark Sold Out
        </button>
        <button
          class="sc-gtsrHT sc-fujyAs blUuCa gJOeZs"
          color="#db9487"
        >
          Delete
        </button>
      </div>
    </div>
    <img
      alt="Sale Item"
      class="sc-jrsJWt eFRJSZ"
      src="http://localhost:5000/items/item/3/image"
    />
  </div>
</DocumentFragment>
`;

exports[`matches snapshot, showSoldOut false, isSold true 1`] = `
<DocumentFragment>
  <div
    class="sc-hKFxyN sc-eCApnc kpxIQl inHhUO"
  >
    <div
      class="sc-kEqXSa bXyTKs"
    >
      <p
        class="sc-dlnjwi sc-jSFjdj sc-gKAaRy dHRUYW hDOwCU kzLjJL"
      >
        Test Name 4
      </p>
      <p
        class="sc-dlnjwi sc-jSFjdj dHRUYW hDOwCU"
      />
      <p
        class="sc-dlnjwi sc-jSFjdj sc-gKAaRy dHRUYW hDOwCU kzLjJL"
      >
        Sold Out!
      </p>
      <p
        class="sc-dlnjwi sc-jSFjdj dHRUYW hDOwCU"
      >
        Price: $456.66
      </p>
      <p
        class="sc-dlnjwi sc-jSFjdj dHRUYW hDOwCU"
      >
        Shipping: $678.44
      </p>
      <p
        class="sc-dlnjwi sc-jSFjdj dHRUYW hDOwCU"
      >
        Total: $1135.10
      </p>
      <div
        class="sc-iCoGMd gCnnPv"
      >
        <button
          class="sc-gtsrHT qvdyu"
          color="#a1c9c9"
        >
          Edit
        </button>
        <button
          class="sc-bdnxRM sc-pNWdM gcBejj kbzcVe"
        >
          Mark Sold Out
        </button>
        <button
          class="sc-gtsrHT sc-fujyAs blUuCa gJOeZs"
          color="#db9487"
        >
          Delete
        </button>
      </div>
    </div>
    <img
      alt="Sale Item"
      class="sc-jrsJWt eFRJSZ"
      src="http://localhost:5000/items/item/4/image"
    />
  </div>
</DocumentFragment>
`;

exports[`matches snapshot, showSoldOut true, isSold false 1`] = `
<DocumentFragment>
  <div
    class="sc-hKFxyN sc-eCApnc kpxIQl inHhUO"
  >
    <div
      class="sc-kEqXSa bXyTKs"
    >
      <p
        class="sc-dlnjwi sc-jSFjdj sc-gKAaRy dHRUYW hDOwCU kzLjJL"
      >
        Test Name
      </p>
      <p
        class="sc-dlnjwi sc-jSFjdj dHRUYW hDOwCU"
      />
      <p
        class="sc-dlnjwi sc-jSFjdj dHRUYW hDOwCU"
      >
        Quantity: 3
      </p>
      <p
        class="sc-dlnjwi sc-jSFjdj dHRUYW hDOwCU"
      >
        Price: $299.99
      </p>
      <p
        class="sc-dlnjwi sc-jSFjdj dHRUYW hDOwCU"
      >
        Shipping: $123.99
      </p>
      <p
        class="sc-dlnjwi sc-jSFjdj dHRUYW hDOwCU"
      >
        Total: $423.98
      </p>
      <div
        class="sc-iCoGMd gCnnPv"
      >
        <button
          class="sc-gtsrHT qvdyu"
          color="#a1c9c9"
        >
          Edit
        </button>
        <button
          class="sc-bdnxRM sc-pNWdM gcBejj eHluva"
        >
          Mark Sold Out
        </button>
        <button
          class="sc-gtsrHT sc-fujyAs blUuCa gJOeZs"
          color="#db9487"
        >
          Delete
        </button>
      </div>
    </div>
    <img
      alt="Sale Item"
      class="sc-jrsJWt eFRJSZ"
      src="http://localhost:5000/items/item/1/image"
    />
  </div>
</DocumentFragment>
`;

exports[`matches snapshot, showSoldOut true, isSold true 1`] = `
<DocumentFragment>
  <div
    class="sc-hKFxyN sc-eCApnc kpxIQl inHhUO"
  >
    <div
      class="sc-kEqXSa bXyTKs"
    >
      <p
        class="sc-dlnjwi sc-jSFjdj sc-gKAaRy dHRUYW hDOwCU kzLjJL"
      >
        Test Name 2
      </p>
      <p
        class="sc-dlnjwi sc-jSFjdj dHRUYW hDOwCU"
      />
      <p
        class="sc-dlnjwi sc-jSFjdj sc-gKAaRy dHRUYW hDOwCU kzLjJL"
      >
        Sold Out!
      </p>
      <p
        class="sc-dlnjwi sc-jSFjdj dHRUYW hDOwCU"
      >
        Price: $299.99
      </p>
      <p
        class="sc-dlnjwi sc-jSFjdj dHRUYW hDOwCU"
      >
        Shipping: $123.99
      </p>
      <p
        class="sc-dlnjwi sc-jSFjdj dHRUYW hDOwCU"
      >
        Total: $423.98
      </p>
      <div
        class="sc-iCoGMd gCnnPv"
      >
        <button
          class="sc-gtsrHT qvdyu"
          color="#a1c9c9"
        >
          Edit
        </button>
        <button
          class="sc-bdnxRM sc-pNWdM gcBejj eHluva"
        >
          Mark Sold Out
        </button>
        <button
          class="sc-gtsrHT sc-fujyAs blUuCa gJOeZs"
          color="#db9487"
        >
          Delete
        </button>
      </div>
    </div>
    <img
      alt="Sale Item"
      class="sc-jrsJWt eFRJSZ"
      src="http://localhost:5000/items/item/2/image"
    />
  </div>
</DocumentFragment>
`;
